ReactWindows/ReactNative/Views/View/ReactViewManager.cs

```diff
@@ -0,0 +1,270 @@
+                            }
+                        }
+                    },
+                };
+            }
+        }
+
+        /// <summary>
+        /// Enables the Canvas as a drop target.
+        /// </summary>
+        [ReactProp("allowDrop")]
+        public void SetAllowDrop(BorderedCanvas view, bool allowDrop)
+        {
+            view.AllowDrop = allowDrop;
+
+            if (allowDrop)
+            {
+                view.DragEnter += OnDragEnter;
+                view.DragOver += OnDragOver;
+                view.Drop += OnDrop;
+                view.DragLeave += OnDragLeave;
+            }
+            else
+            {
+                view.DragEnter -= OnDragEnter;
+                view.DragOver -= OnDragOver;
+                view.Drop -= OnDrop;
+                view.DragLeave -= OnDragLeave;
+            }
+        }
+
+        private static async Task<JObject> GetDataTransferInfo(DataPackageView data, bool drop = false)
+        {
+            var files = new JArray();
+            var items = new JArray();
+            var types = new JArray();
+
+            if (data.Contains(StandardDataFormats.StorageItems))
+            {
+                foreach (var item in await data.GetStorageItemsAsync())
+                {
+                    try
+                    {
+                        var file = item as StorageFile;
+
+                        if (file != null)
+                        {
+                            var props = await file.GetBasicPropertiesAsync();
+                            var type = file.ContentType;
+                            var path = drop ? "urn:future-access-list:" + Uri.EscapeUriString(GetAccessToken(file)) : "";
+
+                            files.Add(new JObject
+                            {
+                                { "name", file.Name },
+                                { "size", props.Size },
+                                { "type", type },
+                                { "uri", path }
+                            });
+
+                            items.Add(new JObject
+                            {
+                                { "kind", "file" },
+                                { "type", type }
+                            });
+
+                            types.Add(type);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        System.Diagnostics.Debug.WriteLine("GetDataTransferInfo: " + ex);
+                    }
+                }
+            }
+
+            return new JObject
+            {
+                { "files", files },
+                { "items", items },
+                { "types", types }
+            };
+        }
+
+        private static string GetAccessToken(StorageFile file)
+        {
+            var futureAccessList = StorageApplicationPermissions.FutureAccessList;
+            var existingEntries = futureAccessList.Entries;
```

Comment:
Do you think it makes sense to make a native module for the FutureAccessList so JS users can query / modify it directly as well?

I think in general, we're also going to need an abstraction to resolve URIs, like the one you generate in the GetDataTransferInfo, in case you need to use it, e.g., as a payload for the NetworkingModule, etc. We're actually going to need something similar for URI resolution for the BlobModule when we implement it. E.g., Android has a URI registration mechanism, but I'm not sure if there's something similar for Windows.

Reply:
We need something like Content Provider in Android, but afaik UWP doesn't have such thing.

JS can't do much with these "future access list" URIs. We could give it API to read file contents, but I doubt that it will be practical to transfer 100 MB files to JS. In S4L we have an RN module that moves/copies/deletes files and there I had to implement handling for FutureAccessList. S4L also has a module to upload/download files and there I also had to handle these URIs.

My take on that is we do need a native module, but that module will be used by 3rd party RN modules, not by JS. In other words:

1. An RN module (or RNW itself) generates a special "content" URI to represent a file.
2. JS gets the URI, but only to give it to another RN module.
3. That other RN module gets the URI and uses the RNW API to resolve it back to the file object.

Reply:
For now I'll add a simple FutureAccessList class (only to make code a bit more organized). Later it'll be replaced with something like a general "content provider" class.